albumentations/augmentations/transforms.py

```diff
@@ -912,6 +912,53 @@ def get_transform_init_args_names(self):
                 'mask_value', 'approximate')
 
 
+class RandomGridShuffle(DualTransform):
+    """
+    Random shuffle grid's cells on image.
+
+    Args:
+        grid ((int, int)): size of grid for splitting image.
+
+    Targets:
+        image, mask
+
+    Image types:
+        uint8, float32
+    """
+
+    def __init__(self, grid=(3, 3), always_apply=False, p=1.0):
+        super(RandomGridShuffle, self).__init__(always_apply, p)
+        self.grid = grid
+
+    def update_params(self, params, **kwargs):
+        params = super(RandomGridShuffle, self).update_params(params, **kwargs)
+        rows = params['rows']
+        cols = params['cols']
+
+        random_state = random.randint(0, 10000)
+
+        params.update({
+            "bboxes": F.split_and_shuffle_shape_by_grid((rows, cols), self.grid, random_state),
```

Comment:
I would, probably replace

`random_state` by `np.random.RandomState(random_state)`

and avoid setting seed in `split_and_shuffle_state_by_grid`, similar to https://github.com/albu/albumentations/blob/master/albumentations/augmentations/transforms.py#L1854

But I do not have strong opinion.

@albu @BloodAxe @creafz what do you think?

Reply:
I followed the next example: https://github.com/albu/albumentations/blob/ef68cd48cb9743f25846913965e2143a98f062a5/albumentations/augmentations/functional.py#L789